src/coordinator/tsdb/remote/compressed_codecs.go

```diff
@@ -93,11 +94,28 @@ func compressedSegmentsFromReaders(readers xio.ReaderSliceOfSlicesIterator) (*rp
 	return segments, nil
 }
 
-func compressedTagsFromTagIterator(tagIter ident.TagIterator) ([]*rpc.CompressedTag, error) {
-	tags := make([]*rpc.CompressedTag, 0, tagIter.Remaining())
+func compressedTagsFromTagIteratorWithEncoder(tagIter ident.TagIterator, iterPools encoding.IteratorPools) ([]byte, bool) {
+	if iterPools == nil {
+		return nil, false
+	}
+	encoder := iterPools.TagEncoder().Get()
```

Comment:
nil check ?

Reply:
That implies a failing in the iterPools, but I'll add a sanity check